runtime/registry.go

```diff
@@ -61,7 +64,16 @@ func (r *Runtime) WaitUntilIdle(ctx context.Context, instanceID string, ignoreHi
 
 // CreateInstance creates a new instance and starts a controller for it.
 func (r *Runtime) CreateInstance(ctx context.Context, inst *drivers.Instance) error {
-	return r.registryCache.create(ctx, inst)
+	if err := r.registryCache.create(ctx, inst); err != nil {
+		return err
+	}
+
+	if r.opts.DataDir != "" {
+		if err := os.Mkdir(filepath.Join(r.opts.DataDir, inst.ID), os.ModePerm); err != nil && !errors.Is(err, fs.ErrExist) {
+			return err
+		}
+	}
```

Comment:
It would be better to do this in `registryCache.add` to make it self-recovering in case the directory is removed.